explorer/explorerroutes.go

```diff
@@ -1101,6 +1101,22 @@ func (exp *explorerUI) TxPage(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// attempt to read swap info assuming this is a swap redemption tx
+	var swapInfo *dbtypes.AtomicSwapInfo
+	swapInfo, err = exp.dataSource.GetAtomicSwapInfoFromRedemptionTx(tx.TxID)
```

Comment:
Unlike `appContext`, `explorerUI` doesn't have a node client, so it would have to use the dataSource wrapper.

Reply:
`GetAtomicSwapInfoFromRedemptionTx`, unlike `GetAtomicSwapInfo` (used in `appContext`) does not read from the db at all, so instead of keeping the dataSource method (even though it would only be a wrapper around the rpcutils function), I added a `dataSource.NodeClient()` method similar to the `dataSource.GetChainParams()` method to give `explorerUI` access to the node client.

I believe we can even do away with both `dataSource.GetChainParams()` and `dataSource.NodeClient()` methods as
- neither function performs any db work AND
- `explorerUI` can get those objects directly when initialized from `ExplorerConfig` rather than from `dataSource`; as is similarly done for `appContext` which is currently how `appContext` gets its node client.